composer/models/mmdetection.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""A wrapper class that converts mmdet detection models to composer models"""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, List, Optional
+
+import numpy as np
+import torch
+from torchmetrics import Metric
+from torchmetrics.collections import MetricCollection
+
+from composer.models import ComposerModel
+
+if TYPE_CHECKING:
+    import mmdetection as mmdet
+
+__all__ = ['MMDetModel']
+
+
+class MMDetModel(ComposerModel):
+    """
+    A wrapper class that adapts mmdetection detectors to composer models.
+
+    Args:
+        model (mmdet.models.detectors.BaseDetector): An MMdetection Detector
+        metrics (list[Metric], optional): list of torchmetrics to apply to the output of `eval_forward`. Default: ``None``.
+
+    .. warning:: This wrapper is designed to work with mmdet datasets.
+
+    Example:
+
+    .. testcode::
+
+        from mmdet.models import build_model
+        from mmcv import ConfigDict
+        from composer.models import MMDetModel
+
+        yolox_s_config = dict(
+            type='YOLOX',
+            input_size=(640, 640),
+            random_size_range=(15, 25),
+            random_size_interval=10,
+            backbone=dict(type='CSPDarknet', deepen_factor=0.33, widen_factor=0.5),
+            neck=dict(type='YOLOXPAFPN', in_channels=[128, 256, 512], out_channels=128, num_csp_blocks=1),
+            bbox_head=dict(type='YOLOXHead', num_classes=num_classes, in_channels=128, feat_channels=128),
```

Comment:
Would the doc tests fail if num_classes is not specified? Same with the composer model documentation